pkg/controller/certificaterequests/sync_test.go

```diff
@@ -140,12 +143,30 @@ func TestSync(t *testing.T) {
 			LastTransitionTime: &nowMetaTime,
 		}),
 	)
+	baseCRNotApprovedEC := gen.CertificateRequest("test-cr",
+		gen.SetCertificateRequestIsCA(false),
+		gen.SetCertificateRequestCSR(csrECPEM),
+		gen.SetCertificateRequestIssuer(cmmeta.ObjectReference{
+			Kind: baseIssuer.Kind,
+			Name: baseIssuer.Name,
+		}),
+	)
+
+	baseCREC := gen.CertificateRequestFrom(baseCRNotApprovedEC,
+		gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+			Type:               cmapi.CertificateRequestConditionApproved,
+			Status:             cmmeta.ConditionTrue,
+			Reason:             "cert-manager.io",
+			Message:            "Certificate request has been approved by cert-manager.io",
+			LastTransitionTime: &nowMetaTime,
+		}),
+	)
 
 	certRSAPEM := generateSelfSignedCert(t, baseCR, skRSA, fixedClockStart, fixedClockStart.Add(time.Hour*12))
 	certRSAPEMExpired := generateSelfSignedCert(t, baseCR, skRSA, fixedClockStart.Add(-time.Hour*13), fixedClockStart.Add(-time.Hour*12))
 
-	certECPEM := generateSelfSignedCert(t, baseCR, skEC, fixedClockStart, fixedClockStart.Add(time.Hour*12))
-	certECPEMExpired := generateSelfSignedCert(t, baseCR, skEC, fixedClockStart.Add(-time.Hour*13), fixedClockStart.Add(-time.Hour*12))
+	certECPEM := generateSelfSignedCert(t, baseCREC, skEC, fixedClockStart, fixedClockStart.Add(time.Hour*12))
+	certECPEMExpired := generateSelfSignedCert(t, baseCREC, skEC, fixedClockStart.Add(-time.Hour*13), fixedClockStart.Add(-time.Hour*12))
```

Comment:
This bit had an error that wasn't caught before- the same CSR generated for the RSA key was used to create the ECDSA PEM. This would now failed when `generateSelfSignedCert` calls `x509.CreateCertificate` as in Go 1.17 this function verifies that the public key matches the private key https://golang.org/doc/go1.17#minor_library_changes

Reply:
Great.